src/core/crypto_provider.rs

```diff
@@ -0,0 +1,27 @@
+//! # Crypto provider module
+//!
+//! This module contains the [`CryptoProvider`] trait, which is used to
+//! implement a module that can be used to configure [`PubNubClientInstance`] or
+//! for manual data encryption and decryption.
+
+use crate::{
+    core::PubNubError,
+    lib::{alloc::vec::Vec, core::fmt::Debug},
+};
+
+/// Crypto provider trait.
+pub trait CryptoProvider: Debug + Send + Sync {
```

Comment:
Do we need that `Debug` here? Do we print any debug data for Crypto Provider?

Reply:
@Xavrax our `PubNubClientInstance` derive from Debug – and this is where everything began :D

Reply:
Makes sense.